src/ansys/openapi/common/_oidc.py

```diff
@@ -149,14 +151,19 @@ def get_session_with_provided_token(self, refresh_token: str) -> requests.Sessio
         except InvalidGrantRequest as excinfo:
             logger.debug(str(excinfo))
             raise ValueError("The provided refresh token was invalid, please request a new token.")
-        with OAuth2.token_cache.forbid_concurrent_missing_token_function_call:
+        try:
```

Comment:
I have done a reasonable amount of bending-over-backwards here to support older versions of the package. Instead of this, we could just raise the minimum version to 8.0.0

Reply:
Looks like even requests-auth 8.0.0 supports python 3.8, I think we can just bump the minimum version (it only depends on requests 2)

Reply:
My only concern was if some other package someone was using with a PyGranta package also depends on requests-auth and _hasn't_ been updated. I think that's unlikely though.